### PR TITLE
camelCase embeds

### PR DESCRIPTION
**Pull request by Nuckyz**

### CODE REVIEW

File: lib/types/channels.d.ts [hunk 1]
```diff
@@ -239,88 +239,168 @@ export interface CreateMessageOptions {
...
+export interface RawEmbedFooterOptionsBase {
     text: string;
 }
-export interface EmbedImageOptions {
+export interface RawEmbedFooterOptions extends RawEmbedFooterOptionsBase {
+    icon_url?: string;
+}
+
+export interface EmbedFooterOptions extends RawEmbedFooterOptionsBase {
+    iconUrl?: string;
```
Comment (DonovanDMC) [inline lib/types/channels.d.ts]:
```suggestion
    iconURL?: string;
```

File: lib/types/channels.d.ts [hunk 2]
```diff
@@ -239,88 +239,168 @@ export interface CreateMessageOptions {
...
     url?: string;
 }
 
-export interface EmbedField {
+export interface RawEmbedAuthorOptions extends RawEmbedAuthorOptionsBase {
+    icon_url?: string;
+}
+
+export interface EmbedAuthorOptions extends RawEmbedAuthorOptionsBase {
+    iconUrl?: string;
```
Comment (DonovanDMC) [inline lib/types/channels.d.ts]:
```suggestion
    iconURL?: string;
```

File: lib/types/channels.d.ts [hunk 3]
```diff
@@ -239,88 +239,168 @@ export interface CreateMessageOptions {
...
+
+export type EmbedField = RawEmbedField;
+
+export interface RawEmbedFooter extends RawEmbedFooterOptions {
     proxy_icon_url?: string;
 }
-export interface EmbedImage extends EmbedImageOptions {
+
+export interface EmbedFooter extends EmbedFooterOptions {
+    proxyIconUrl?: string;
```
Comment (DonovanDMC) [inline lib/types/channels.d.ts]:
```suggestion
    proxyIconURL?: string;
```

File: lib/types/channels.d.ts [hunk 4]
```diff
@@ -239,88 +239,168 @@ export interface CreateMessageOptions {
...
-export interface EmbedThumbnail extends EmbedThumbnailOptions {
-    height?: number;
+export interface RawEmbedImage extends RawEmbedImageBase, RawEmbedImageOptions {
     proxy_url?: string;
-    width?: number;
 }
 
-export interface EmbedVideo {
+export interface EmbedImage extends RawEmbedImageBase, EmbedImageOptions {
+    proxyUrl?: string;
```
Comment (DonovanDMC) [inline lib/types/channels.d.ts]:
```suggestion
    proxyURL?: string;
```

File: lib/types/channels.d.ts [hunk 5]
```diff
@@ -239,88 +239,168 @@ export interface CreateMessageOptions {
...
     height?: number;
+    width?: number;
+}
+
+export interface RawEmbedThumbnail extends RawEmbedThumbnailBase, RawEmbedThumbnailOptions {
     proxy_url?: string;
+}
+
+export interface EmbedThumbnail extends RawEmbedThumbnailBase, EmbedThumbnailOptions {
+    proxyUrl?: string;
```
Comment (DonovanDMC) [inline lib/types/channels.d.ts]:
```suggestion
    proxyURL?: string;
```

File: lib/types/channels.d.ts [hunk 6]
```diff
@@ -239,88 +239,168 @@ export interface CreateMessageOptions {
...
     width?: number;
 }
 
-export interface EmbedProvider {
+export interface RawEmbedVideo extends RawEmbedVideoBase {
+    proxy_url?: string;
+}
+
+export interface EmbedVideo extends RawEmbedVideoBase {
+    proxyUrl?: string;
```
Comment (DonovanDMC) [inline lib/types/channels.d.ts]:
```suggestion
    proxyURL?: string;
```

File: lib/types/channels.d.ts [hunk 7]
```diff
@@ -239,88 +239,168 @@ export interface CreateMessageOptions {
...
-export interface EmbedAuthor extends EmbedAuthorOptions {
+
+export type EmbedProvider = RawEmbedProvider;
+
+export interface RawEmbedAuthor extends RawEmbedAuthorOptions {
     proxy_icon_url?: string;
 }
 
+export interface EmbedAuthor extends EmbedAuthorOptions {
+    proxyIconUrl?: string;
```
Comment (DonovanDMC) [inline lib/types/channels.d.ts]:
```suggestion
    proxyIconURL?: string;
```

File: lib/util/Util.ts [hunk 8]
```diff
@@ -137,6 +143,107 @@ export default class Util {
...
     }
 
+    embedsToParsed(embeds: Array<RawEmbed>): Array<Embed> {
+        return embeds.map((embed) => {
+            const parsedEmbed: Embed = {};
+
+            if (embed.author) {
+                parsedEmbed.author = {
+                    name:         embed.author.name,
+                    iconUrl:      embed.author.icon_url,
```
Comment (DonovanDMC) [inline lib/util/Util.ts]:
```suggestion
                    iconURL:      embed.author.icon_url,
```

File: lib/util/Util.ts [hunk 9]
```diff
@@ -137,6 +143,107 @@ export default class Util {
...
+                    proxyIconUrl: embed.author.proxy_icon_url,
```
Comment (DonovanDMC) [inline lib/util/Util.ts]:
```suggestion
                    proxyIconURL: embed.author.proxy_icon_url,
```

File: lib/util/Util.ts [hunk 10]
```diff
@@ -137,6 +143,107 @@ export default class Util {
...
+                parsedEmbed.fields = embed.fields.map((field): EmbedField => ({
+                    inline: field.inline,
+                    name:   field.name,
+                    value:  field.value
+                }));
+            }
+            if (embed.footer) {
+                parsedEmbed.footer = {
+                    text:         embed.footer.text,
+                    iconUrl:      embed.footer.icon_url,
```
Comment (DonovanDMC) [inline lib/util/Util.ts]:
```suggestion
                    iconURL:      embed.footer.icon_url,
```

File: lib/util/Util.ts [hunk 11]
```diff
@@ -137,6 +143,107 @@ export default class Util {
...
+                    proxyIconUrl: embed.footer.proxy_icon_url
```
Comment (DonovanDMC) [inline lib/util/Util.ts]:
```suggestion
                    proxyIconURL: embed.footer.proxy_icon_url
```

File: lib/util/Util.ts [hunk 12]
```diff
@@ -137,6 +143,107 @@ export default class Util {
...
+                };
+            }
+            if (embed.timestamp) parsedEmbed.timestamp = embed.timestamp;
+            if (embed.title) parsedEmbed.title = embed.title;
+            if (embed.image) {
+                parsedEmbed.image = {
+                    url:      embed.image.url,
+                    height:   embed.image.height,
+                    proxyUrl: embed.image.proxy_url,
```
Comment (DonovanDMC) [inline lib/util/Util.ts]:
```suggestion
                    proxyURL: embed.image.proxy_url,
```

File: lib/util/Util.ts [hunk 13]
```diff
@@ -137,6 +143,107 @@ export default class Util {
...
+                };
+            }
+            if (embed.provider) parsedEmbed.provider = {
+                name: embed.provider.name,
+                url:  embed.provider.url
+            };
+            if (embed.thumbnail) parsedEmbed.thumbnail = {
+                url:      embed.thumbnail.url,
+                height:   embed.thumbnail.height,
+                proxyUrl: embed.thumbnail.proxy_url,
```
Comment (DonovanDMC) [inline lib/util/Util.ts]:
```suggestion
                proxyURL: embed.thumbnail.proxy_url,
```

File: lib/util/Util.ts [hunk 14]
```diff
@@ -137,6 +143,107 @@ export default class Util {
...
+                width:    embed.thumbnail.width
+            };
+            if (embed.url) parsedEmbed.url = embed.url;
+            if (embed.type) parsedEmbed.type =  embed.type;
+            if (embed.video) {
+                parsedEmbed.video = {
+                    height:   embed.video.height,
+                    proxyUrl: embed.video.proxy_url,
```
Comment (DonovanDMC) [inline lib/util/Util.ts]:
```suggestion
                    proxyURL: embed.video.proxy_url,
```

File: lib/util/Util.ts [hunk 15]
```diff
@@ -137,6 +143,107 @@ export default class Util {
         return img;
     }
 
+    embedsToParsed(embeds: Array<RawEmbed>): Array<Embed> {
+        return embeds.map((embed) => {
```
Comment (DonovanDMC) [inline lib/util/Util.ts]:
```suggestion
        return embeds.map(embed => {
```

File: lib/util/Util.ts [hunk 16]
```diff
@@ -137,6 +143,107 @@ export default class Util {
...
+                    width:    embed.video.width
+                };
+            }
+
+            return parsedEmbed;
+        });
+    }
+
+    embedsToRaw(embeds: Array<EmbedOptions>): Array<RawEmbedOptions> {
+        return embeds.map((embed) => {
```
Comment (DonovanDMC) [inline lib/util/Util.ts]:
```suggestion
        return embeds.map(embed => {
```

File: lib/util/Util.ts [hunk 17]
```diff
@@ -137,6 +143,107 @@ export default class Util {
...
+            const rawEmbed: RawEmbedOptions = {};
+
+            if (embed.author) {
+                rawEmbed.author = {
+                    name:     embed.author.name,
+                    icon_url: embed.author.iconUrl,
```
Comment (DonovanDMC) [inline lib/util/Util.ts]:
```suggestion
                    icon_url: embed.author.iconURL,
```

File: lib/util/Util.ts [hunk 18]
```diff
@@ -137,6 +143,107 @@ export default class Util {
...
+                    url:      embed.video.url,
+                    width:    embed.video.width
+                };
+            }
+
+            return parsedEmbed;
+        });
+    }
+
+    embedsToRaw(embeds: Array<EmbedOptions>): Array<RawEmbedOptions> {
```
Comment (DonovanDMC) [inline lib/util/Util.ts:207]: missing footer

**Review by DonovanDMC — CHANGES_REQUESTED** [review]

File: lib/util/Util.ts (hunk 18 above)
Comment (Nuckyz) [reply lib/util/Util.ts:207]: done

**Comment by DonovanDMC** [comment]
I've done 213fd30487a6be6f6ea64a52c58e88bda9215b01 which dedupes various interfaces and properties, I want someone else to run through it to make sure I didn't fuck everything up

**Review by DonovanDMC — APPROVED** [review]

**Pull request merged**